cmd/github.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/urfave/cli/v2"
+)
+
+// lookupGitHubToken will try to find a GitHub access token that can then be used for various API calls but also cloning of private repositories.
+func lookupGitHubToken(cctx *cli.Context) (string, error) {
+	log.Print("Looking for a GitHub token")
+	token := cctx.String("github-token")
+	if token != "" {
+		log.Print("Using GitHub token provided via flag")
+		return token, nil
+	}
+	// First try: Check if it's in the environment. This can override everything!
+	token = os.Getenv("GITHUB_TOKEN")
+	if token != "" {
+		log.Print("Using GitHub token provided via environment variable")
+		return token, nil
+	}
```

Comment:
I like this and I was thinking about this earlier, how we should prioritize this over mounting the 'gh' directory or trying to use the 'gh' command.

Reply:
Among others. It's also the environment variable that would be picked up by `gh` if it's set and pretty much all other tooling that somehow interacts with GitHub.